lib/matplotlib/ticker.py

```diff
@@ -2468,22 +2468,23 @@ def tick_values(self, vmin, vmax):
         # "simple" mode is when the range falls entirely within (-t,
         # t) -- it should just display (vmin, 0, vmax)
 
+        # Determine which of the three ranges we have
```

Comment:
I think the whole thing can be simplified to
```
if -linthresh <= vmin <= vmax <= 0 or 0 <= vmin <= vmax <= linthresh:
    return [vmin, vmax]
has_a = (vmin < -linthresh)
has_b = (-linthresh < vmin < linthresh or -linthresh < vmax < linthresh)
has_c = (linthresh < vmax)
```

Reply:
I think `SymLogNorm` needs quite a few more tests, which I plan to add in for `3.2` at some point; I think this is a good suggestion, but might save it for after I've written tests that exercise this logic properly.

Reply:
sure